lib/pbench/server/api/resources/users_api.py

```diff
@@ -326,36 +370,24 @@ def get(self, username):
                     }
         """
         try:
-            user, verified = self.auth.verify_user(username)
-        except Exception:
-            self.logger.exception("Exception occurred during verifying the user")
-            abort(500, message="INTERNAL ERROR")
-
-        # TODO: Check if the user has the right privileges
-        if verified:
-            response_object = user.get_json()
-            return make_response(jsonify(response_object), 200)
-        elif user.is_admin():
-            try:
-                target_user = User.query(username=username)
-                response_object = target_user.get_json()
-                return make_response(jsonify(response_object), 200)
-            except Exception:
-                self.logger.exception(
-                    "Exception occurred while querying the user. Username: {}", username
-                )
-                abort(500, message="INTERNAL ERROR")
-        else:
-            self.logger.warning(
-                "User {} is not authorized to get user {}.", user.username, username,
+            target_user, http_status, http_message = self.get_valid_target_user(
+                target_username, "GET"
```

Comment:
While this is, again, syntactically correct, doing it this way removes all the advantages of using a named tuple.  Instead, I think it would be better to do something like this:
```
            result = self.get_valid_target_user(target_username, "GET")
[...]
        if result.user is None:
            abort(result.http_status, message=result.http_message)
```

Reply:
Ditto in PUT and DELETE.